### PR TITLE
842 replace db question confirmation

### PR DESCRIPTION
**Pull request by adammahmood**
### What is the context of this PR?
- Fixes question confirmation queries and mutations
- Adds a confirm modal when deleting a question confirmation

### How to review 
- Ensure questionnaire JSON is updated as expected
- Ensure queries return expected result
- Ensure confirm modal appears when deleting a question confirmation

### CODE REVIEW

**Comment by adammahmood** [comment]
[QuestionConfirmation](https://trello.com/c/VPm9wHx7/842-questionconfirmation)

File: eq-author-api/schema/resolvers/base.js [hunk 1]
```diff
@@ -1076,18 +1071,9 @@ const Resolvers = {
...
-    negative: ({ negativeLabel, negativeDescription }) => ({
-      label: negativeLabel,
-      description: negativeDescription,
-    }),
     availablePipingAnswers: ({ id }, args, ctx) =>
-      ctx.repositories.QuestionConfirmation.getPipingAnswers(id),
-    availablePipingMetadata: ({ id }, args, ctx) =>
-      ctx.repositories.QuestionConfirmation.getPipingMetadata(id),
+      getPreviousAnswersForPage(ctx.questionnaire, id),
+    availablePipingMetadata: (page, args, ctx) => ctx.questionnaire.metadata,
```
Comment (tgandrews) [inline eq-author-api/schema/resolvers/base.js]: Are these filtered by type? Or can we pipe all metadata?

File: eq-author/src/App/questionConfirmation/Design/index.js [hunk 2]
```diff
@@ -27,14 +29,31 @@ export class UnwrappedQuestionConfirmationRoute extends React.Component {
...
+    showDeleteConfirmDialog: false,
+  };
+
+  handleOpenDeleteConfirmDialog = () =>
+    this.setState({ showDeleteConfirmDialog: true });
+
+  handleCloseDeleteConfirmDialog = cb =>
+    this.setState(
+      { showDeleteConfirmDialog: false },
+      isFunction(cb) ? cb : null
```
Comment (tgandrews) [inline eq-author/src/App/questionConfirmation/Design/index.js]: I think `setState` will handle this function check for us. I don't think we need to do it ourselves.

File: eq-author/src/App/questionConfirmation/Design/index.js [hunk 3]
```diff
@@ -27,14 +29,31 @@ export class UnwrappedQuestionConfirmationRoute extends React.Component {
...
     );
+
+  handleDeletePageConfirm = () => {
+    this.handleCloseDeleteConfirmDialog(() => {
```
Comment (tgandrews) [inline eq-author/src/App/questionConfirmation/Design/index.js]: Could we just call setState as the function doesn't seem to do a lot.

File: eq-author-api/schema/resolvers/base.js (hunk 1 above)
Comment (adammahmood) [reply eq-author-api/schema/resolvers/base.js]: You can pipe any metadata into a question confirmation.

File: eq-author/src/App/questionConfirmation/Design/index.js (hunk 2 above)
Comment (adammahmood) [reply eq-author/src/App/questionConfirmation/Design/index.js]: `cb` isn't always a function and `setState` will throw `Error: Invalid argument passed as callback. Expected a function. Instead received: ...` when an invalid type is passed as a callback.

File: eq-author/src/App/questionConfirmation/Design/index.js (hunk 3 above)
Comment (adammahmood) [reply eq-author/src/App/questionConfirmation/Design/index.js]: I can set state in the separate functions. 👍

File: eq-author/src/App/questionConfirmation/Design/index.js (hunk 2 above)
Comment (tgandrews) [reply eq-author/src/App/questionConfirmation/Design/index.js]: I was expecting to be `undefined` is that not the case?

Comment (tgandrews) [reply eq-author/src/App/questionConfirmation/Design/index.js]: No longer applicable.

**Comment by adammahmood** [comment]
> Looks good just a couple of questions. Also, I was expecting a cypress change for undelete?

@tgandrews I've fixed the cypress tests to work with the latest UX.  I've decided not to touch anything related to "undo" / "undelete" as this is being stripped out in a separate PR.

**Pull request merged**